intro_to_rl.ipynb

```diff
@@ -0,0 +1,3225 @@
+{
```

Comment:
<p>People usually use discount factors to allow proofs for MDPs by preventing infinite returns, which seems like a rather technical point at this stage (and yes you could use it for other purposes. But is this important for a first example of RL? Its much conceptually clearer just saying 'we want to keep the cartpole from falling over for as long as possible!')</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2022/pull/6/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Reply:
<p>So technically, it allows the return to be bounded when using infinite horizons, but inititaive answers to this include it would allow you weight how much the agents cares about rewards in the distant future vs immediate future.&nbsp;People could think about what would happen if discount=0 vs 1. I think it is relevant question and important question. </p>